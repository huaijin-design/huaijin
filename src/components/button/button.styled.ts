import styled, { css } from 'styled-components';
import { mapTheme, mapProp } from '../../utils/styled-map';
import { ButtonProps } from './button';
import { reset, colors, shadow, transition } from '../../themes/variable';

export const componentName = 'HJButton';

const PropColor = css`
  color: ${mapTheme('color', 'color.button')};
  background-color: ${mapTheme('color', 'background.button')};
  ${mapTheme('color', 'shadow.button', shadow)};
`;

const PropGhost = css`
  ${mapProp('ghost')`
    box-shadow: none;
    background: transparent;
    border: 1px solid ${colors.white};
    color: ${colors.white};
    &:hover {
      color: ${colors.dark};
      background-color: ${colors.white};
    }
    &:active {
      color: ${colors.dark};
      background-color: ${colors.white};
    }
  `}
`;

const StyledButton = styled.button.attrs({
  className: componentName,
})<ButtonProps>`
  ${reset}
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);

  min-width: 88px;
  min-height: 32px;
  border-radius: 2px;
  font-size: ${mapTheme('fontSize.button')};
  ${transition}
  &:hover {
    transform: translate3d(0,-2px,0) scale3d(1.04,1.04,1.04);
  }
  &:active {
    transform: translate3d(0,0,0) scale3d(1,1,1);
  }

  ${PropColor}
  ${PropGhost}
`;

export default StyledButton;
