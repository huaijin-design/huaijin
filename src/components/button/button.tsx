import * as React from 'react';
import Button, { componentName } from './button.styled';
import { buttonDefaultTheme } from '../../themes/theme';

export type ButtonType = 'submit' | 'button' | 'reset';
export type ButtonColor = 'default' | 'primary' | 'success' | 'warning' | 'danger';
export interface ButtonProps {
  type?: ButtonType;
  onClick?: React.MouseEventHandler;
  className?: string;

  color?: ButtonColor;
  ghost?: boolean;
  disabled?: boolean;
}

Button.displayName = componentName;

Button.defaultProps = {
  type: 'button',
  color: 'default',
  theme: buttonDefaultTheme,
};

export default Button;
