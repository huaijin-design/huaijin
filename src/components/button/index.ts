export { default as Button } from './button.styled';
