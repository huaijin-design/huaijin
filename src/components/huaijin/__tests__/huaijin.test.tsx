import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { matchers } from 'jest-emotion';
import { Huaijin } from '..';
import { Button } from '../../button';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(matchers);

describe('Huaijin', () => {
  it('normal use of the theme', () => {
    const theme = {
      color: {
        dark: '#000',
      },
    };
    const json = renderer
      .create(
        <Huaijin theme={theme}>
          <Button>theme</Button>
        </Huaijin>,
      )
      .toJSON();
    expect(json).toHaveStyleRule('color', theme.color.dark);
    expect(json).toMatchSnapshot();
  });
});
