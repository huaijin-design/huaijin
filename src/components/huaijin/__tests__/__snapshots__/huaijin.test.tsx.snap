// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Huaijin normal use of the theme 1`] = `
.emotion-0 {
  padding: 0;
  outline: none;
  border: none;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(255,255,255,0);
  font-size: 24px;
  font-weight: 300;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Open Sans','Helvetica Neue',sans-serif;
  line-height: 1.3;
  color: #000;
  background: transparent;
  position: relative;
  margin: 0.3em;
}

.emotion-0 > .hj-button-main {
  position: relative;
  z-index: 2;
}

.emotion-0:after {
  content: '';
  position: absolute;
  z-index: 1;
  top: 60%;
  left: -0.2em;
  right: -0.2em;
  bottom: 0;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  background-color: rgba(255,111,97,0.5);
}

.emotion-0:hover:after {
  top: 0;
}

.emotion-0:active:after {
  background-color: rgba(255,111,97,0.6);
  box-shadow: 0 0 0 3px rgba(255,111,97,0.2);
}

<button
  className="emotion-0"
  type="button"
>
  <div
    className="hj-button-main"
  >
    theme
  </div>
</button>
`;
